### ISSUE TITLE
Difference in projection with estimated and ground truth focal length?

### ISSUE THREAD

**Issue by pixelite1201** [huawei-noah/noah-research#99]
Thanks again for sharing the AGORA finetuning checkpoint. I tried running it on AGORA validation images with estimated focal length and ground truth focal length. But seems like the projection when using the ground truth focal length is worse than the assumed focal length specially for people off center in the image.

I have attached the images below. Only difference that I made in the code is to change the function estimate_focal_length in the utils.py. Do you know what could be the reason for this?

**Assumed focal length:**
![assum_fl](https://user-images.githubusercontent.com/26616807/183066412-ab80e573-b242-47f2-87ab-8550b0573291.jpg)

**Ground truth focal length:**
![gt_focal_length](https://user-images.githubusercontent.com/26616807/183066563-7d2cee1b-04bb-4be0-90e8-b69603e427ba.jpg)

**Comment by zhihaolee** [comment]
Could&nbsp;you&nbsp;tell&nbsp;me&nbsp;the&nbsp;values&nbsp;of&nbsp;the&nbsp;estimated&nbsp;and&nbsp;GT&nbsp;focal&nbsp;lengths?
The&nbsp;GT&nbsp;focal&nbsp;length&nbsp;of&nbsp;this&nbsp;image&nbsp;may&nbsp;be&nbsp;infrequent&nbsp;in&nbsp;the&nbsp;training&nbsp;data.


---Original---
From: ***@***.***&gt;
Date: Fri, Aug 5, 2022 19:19 PM
To: ***@***.***&gt;;
Cc: ***@***.***&gt;;
Subject: [huawei-noah/noah-research] Difference in projection with estimatedand ground truth focal length? (Issue #99)




 
Thanks again for sharing the AGORA finetuning checkpoint. I tried running it on AGORA validation images with estimated focal length and ground truth focal length. But seems like the projection when using the ground truth focal length is worse than the assumed focal length.
 
I have attached the images below. Only difference that I made in the code is to change the function estimate_focal_length in the utils.py. Do you know what could be the reason for this?
 
Assumed focal length:
 
 
Ground truth focal length:
 
 
&mdash;

**Comment by zhihaolee** [comment]
Could you tell me the values of the estimated and GT focal lengths?
The GT focal length of this image may be infrequent in the training data.

**Comment by pixelite1201** [comment]
ground truth focal length: 1920
estimated focal length: 4405

**Comment by zhihaolee** [comment]
A 4K image with the focal length of 1920 is an ultra wide-angle one, whose diagonal-FoV is bigger than 90 degree.
Global rotation estimation errors are more noticeable in the perspective projection with such a big FoV.
I think that's why the result looks worse than that with the estimated focal length. You can put the predicted and GT meshes together to see the actual accuracy, like what I did in the last column of Fig. 5.

**Comment by pixelite1201** [comment]
Good point that the extreme focal length makes the global rotation estimation errors more noticeable. 
But if the ground truth focal length is used during the finetuning experiment, does this mean the keypoint loss while finetuning was high as well? Did you have higher weight for 3d joints/vertex loss and low weight for keypoint loss for the finetuning experiment?

**Comment by zhihaolee** [comment]
1. I didn't notice that, but I think yes, higher 2D keypoint loss for extreme focal lengths.
2. No, the same loss weights.

**Comment by pixelite1201** [comment]
Thanks for clarification.
